### PR TITLE
add support for custom web.xml as it is done for jetty-env file

### PR DESCRIPTION
**Pull request by jeanmi151**
tested and working

### CODE REVIEW

**Comment by jeanpommier** [comment]
Something also is bothering me: the datadir is going to diverge quite a bit if we store there stuff that is deployment-specific. For now, not a big deal, a few additional files for the docker/k8s side. But let's suppose that we also do the same for the ansible deployment style. And at some point, it's a mess and we can't figure out what is deployment-specific andwhat's not

File: templates/geoserver/geoserver-deployment.yaml [hunk 1]
```diff
@@ -50,6 +50,9 @@ spec:
           if [ -f /mnt/georchestra_datadir/geoserver/jetty-env.xml ] ; then
             cp /mnt/georchestra_datadir/geoserver/jetty-env.xml /mnt/geoserver_webapp/WEB-INF/jetty-env.xml ;
           fi ;
+          if [ -f /mnt/georchestra_datadir/geoserver/web.xml ] ; then
```
[the remark this answers is not in the document]
Comment (jeanmi151) [reply templates/geoserver/geoserver-deployment.yaml:53]: hum it already exist for jetty-env custom
We got later on a "fix-permissions-pvc" initcontainer for setting the right permissions

**Comment by fvanderbiest** [comment]
> Something also is bothering me: the datadir is going to diverge quite a bit if we store there stuff that is deployment-specific.

Agreed.

**Review by edevosc2c — APPROVED** [review]
good but only for these changes

**Pull request merged**